sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -167,10 +192,45 @@ case class Or(left: Expression, right: Expression)
       }
     }
   }
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): Code = {
+    val eval1 = left.gen(ctx)
+    val eval2 = right.gen(ctx)
+    s"""
+      ${eval1.code}
+      boolean ${ev.nullTerm} = false;
```

Comment:
No, If any of them is `true`( another one could be null), the result is true. Same to AND, if any of them is `false`, the result is `false`.